### PR TITLE
Add condition to check if ruby version is < 2.7

### LINKED ISSUE #6952: Install on Manjaro Linux requires HOMEBREW_FORCE_VENDOR_RUBY
These questions are N/A as it did not install:

- [ ] ran `brew update` and can still reproduce the problem?
- [ ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)
Install on an up-to-date version of Manjaro Linux which is based upon Arch Linux.

## What happened (include command output)
Unfortunately I hadn't configured a long scrollback yet and lost the output, but it was similar to https://stackoverflow.com/questions/54888582/ruby-cannot-load-such-file-active-support-core-ext-object-blank with that same message `cannot load such file -- active_support/core_ext/object/blank (LoadError)`, except ruby version 2.7.0. If you need it, I can destroy the vendor-provided ruby and try reinstalling.

Looking at the script, I noticed that https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh checks for HOMEBREW_FORCE_VENDOR_RUBY. The command `HOMEBREW_FORCE_VENDOR_RUBY=1 sh -c "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh)"` succeeded.

## What you expected to happen

`==> Installation successful!` message.

I think that at the minimum, a recommendation on https://docs.brew.sh/Homebrew-on-Linux to try HOMEBREW_FORCE_VENDOR_RUBY could help. Another possibility is to fallback to downloading ruby if the installation fails.

## Step-by-step reproduction instructions (by running `brew` commands)
Install Manjaro Linux, install all updates, and then run the script from https://docs.brew.sh/Homebrew-on-Linux

## Output of `brew config` and `brew doctor` commands
N/A, not installed by the time it broke. UPDATE:
brew config:
```
HOMEBREW_VERSION: 2.2.4-2-g23108f3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 23108f3c736030070ca1e4673652c20c503a7d02
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core
Core tap HEAD: bafaf942e6834f22e0c99ac5f1ff57b1541fd599
Core tap last commit: 2 days ago
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CACHE: /home/ben/.cache/Homebrew
HOMEBREW_LOGS: /home/ben/.cache/Homebrew/Logs
CPU: quad-core 64-bit skylake
Homebrew Ruby: 2.6.3 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby
Clang: N/A
Git: 2.25.0 => /home/linuxbrew/.linuxbrew/bin/git
Curl: 7.68.0 => /usr/bin/curl
Kernel: Linux 5.3.18-1-MANJARO x86_64 GNU/Linux
OS: Manjaro Linux (Juhraya)
Host glibc: 2.30
/usr/bin/gcc: 9.2.0
glibc: N/A
gcc: N/A
xorg: N/A
```

brew doctor:
```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: python is symlinked to python3
This will confuse build scripts and in general lead to subtle breakage.
```

### PR DESCRIPTION
**Pull request by emreerhan**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Fixes #6952

### CODE REVIEW

File: Library/Homebrew/utils/ruby.sh [hunk 1]
```diff
@@ -45,7 +46,12 @@ setup-ruby-path() {
         ruby_version_new_enough="$("$HOMEBREW_RUBY_PATH" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e "puts Gem::Version.new(RUBY_VERSION.to_s.dup) >= Gem::Version.new('$minimum_ruby_version')")"
       fi
 
-      if [[ -z "$HOMEBREW_RUBY_PATH" || -n "$HOMEBREW_FORCE_VENDOR_RUBY" || "$ruby_version_new_enough" != "true" ]]
+      if [[ -n "$HOMEBREW_RUBY_PATH" && -z "$HOMEBREW_FORCE_VENDOR_RUBY" ]]
+      then
+          ruby_version_old_enough="$("$HOMEBREW_RUBY_PATH" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e "puts Gem::Version.new(RUBY_VERSION.to_s.dup) < Gem::Version.new('$maximum_ruby_version')")"
```
Comment (sjackman) [inline Library/Homebrew/utils/ruby.sh]: I'm inclined to merge lines 46 and 51 into a single test: `acceptable_ruby_version=…`

File: Library/Homebrew/utils/ruby.sh [hunk 2]
```diff
@@ -6,6 +6,7 @@ setup-ruby-path() {
   # When bumping check if HOMEBREW_MACOS_SYSTEM_RUBY_NEW_ENOUGH (in brew.sh)
   # also needs to be changed.
   local minimum_ruby_version="2.6.3"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh]:
```suggestion
   local required_ruby_version="2.6"
```

File: Library/Homebrew/utils/ruby.sh [hunk 3]
```diff
@@ -6,6 +6,7 @@ setup-ruby-path() {
...
+  local maximum_ruby_version="2.7.0"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh]:
```suggestion
```

File: Library/Homebrew/utils/ruby.sh [hunk 4]
```diff
@@ -45,7 +46,12 @@ setup-ruby-path() {
         ruby_version_new_enough="$("$HOMEBREW_RUBY_PATH" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e "puts Gem::Version.new(RUBY_VERSION.to_s.dup) >= Gem::Version.new('$minimum_ruby_version')")"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh]:
```suggestion
         usable_ruby_version="$("$HOMEBREW_RUBY_PATH" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e "puts Gem::Version.new(RUBY_VERSION.to_s.dup) >= Gem::Version.new('$required_ruby_version')")"
```

and adjust L42 to also use `usable_ruby_version`

File: Library/Homebrew/utils/ruby.sh [hunk 5]
```diff
@@ -45,7 +46,12 @@ setup-ruby-path() {
...
       fi
 
-      if [[ -z "$HOMEBREW_RUBY_PATH" || -n "$HOMEBREW_FORCE_VENDOR_RUBY" || "$ruby_version_new_enough" != "true" ]]
+      if [[ -n "$HOMEBREW_RUBY_PATH" && -z "$HOMEBREW_FORCE_VENDOR_RUBY" ]]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh]:
```suggestion
      if [[ -z "$HOMEBREW_RUBY_PATH" || -n "$HOMEBREW_FORCE_VENDOR_RUBY" || "$usable_ruby_version" != "true" ]]
```

File: Library/Homebrew/utils/ruby.sh [hunk 6]
```diff
@@ -45,7 +46,12 @@ setup-ruby-path() {
...
+      then
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh]:
```suggestion
```

File: Library/Homebrew/utils/ruby.sh (hunk 1 above)
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh]:
```suggestion
```

File: Library/Homebrew/utils/ruby.sh [hunk 7]
```diff
@@ -45,7 +46,12 @@ setup-ruby-path() {
...
+          ruby_version_old_enough="$("$HOMEBREW_RUBY_PATH" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e "puts Gem::Version.new(RUBY_VERSION.to_s.dup) < Gem::Version.new('$maximum_ruby_version')")"
+      fi
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh]:
```suggestion
```

File: Library/Homebrew/utils/ruby.sh [hunk 8]
```diff
@@ -45,7 +46,12 @@ setup-ruby-path() {
...
+
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh]:
```suggestion
```

File: Library/Homebrew/utils/ruby.sh [hunk 9]
```diff
@@ -45,7 +46,12 @@ setup-ruby-path() {
...
+      if [[ -z "$HOMEBREW_RUBY_PATH" || -n "$HOMEBREW_FORCE_VENDOR_RUBY" || "$ruby_version_new_enough" != "true" || "$ruby_version_old_enough" != "true" ]]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh]:
```suggestion
```

**Comment by emreerhan** [comment]
Quick question, the MacOS CircleCI test failed because:
```
GitHub Bad credentials:HOMEBREW_GITHUB_API_TOKEN may be invalid or expired
```

It seems like this isn't an issue on my end, is it?

**Comment by emreerhan** [comment]
Scratch my last comment. Seems like all checks have passed. When I have time later this weekend I'm going to try cloning this branch on a Docker instance w/ Ruby 2.7 installed to see if it works!

**Comment by MikeMcQuaid** [comment]
Thanks so much for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @emreerhan!

**Comment by sjackman** [comment]
Thank you for your first contribution to Homebrew/brew, Emre! 🥇

**Pull request merged**